src/types/function.rs

```diff
@@ -17,56 +14,32 @@ impl PyCFunction {
     /// See [raw_pycfunction] for documentation on how to get the `fun` argument.
     pub fn new_with_keywords<'a>(
         fun: ffi::PyCFunctionWithKeywords,
-        name: &str,
+        name: &'static str,
```

Comment:
It's pretty awkward to expect null-terminated `&str` in a public API like this, taking `CStr` would signal this expectation on the type level.

Reply:
I think it's the same for `doc`... but am willing to avoid this.